### DISCUSSION TITLE
Gcode Scripts versus Continuous Print Scripts

### DISCUSSION THREAD

**Discussion by whopster17** [smartin015/continuousprint#177] [Q&A]
So I've noticed an issue and I'm not sure if there's a simple solution that I seem to be missing. Previously, I had a script in the Octoprint GCODE script "Before print job starts". This had a heating purpose and also ran auto bed levelling. However, with the way the new events and scripts work, once the cooldown script starts, it runs this bed levelling and heating script because it sees it as a print starting. This breaks the whole continuous print for me (actually ruined one of my prints).

To solve this issue, I cut the "Before print job starts" script into a custom start script in the continuous print interface and appended it to the "Print Start" event in the continuous print plugin. This solved most of the issue as the start script doesn't run before every sub event in continuous print now, however, I am not able to run the bed levelling sequence now for regular prints that are not in plugin. Is there a functionality I am missing? I am on version 2.3.0rc2.

**Comment by smartin015** [comment]
Hey @whopster17 - you're in the same boat as other folks that use the vanilla settings scripts, in that they're run for every "print file", and CPQ writes scripts to files before running them for a couple different reasons.  

If you want to follow along / provide feedback on #175, that's where I'll be putting updates on improving this interaction between OctoPrint and CPQ.

**Comment by rogiepoo** [comment]
_once the cooldown script starts, it runs this bed levelling_ 
Prusaslicer? "G28" is a command that auto-homes but also INCLUDES bed leveling! (I was shocked to learn; searched for hours/days/weeks), so you may need to replace "G 28" with "G28 W" which does not include autoleveling.  Not sure if this helps (~1 yr later) but thought I'd share my recent discovery.
rr

**Reply by rogiepoo** [reply]
sorry, meant to type "G28"

**Reply by whopster17** [reply]
That is very useful information! Fortunately, I've moved onto the Bambu Lab printers and I don't have this issue anymore (but I did lose the automation ability), but this will no doubt be useful to someone!

**Comment by rogiepoo** [comment]
:) Cheers!

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***
